content/languages/lambdacalc/lc-test.md

```diff
@@ -0,0 +1,114 @@
+---
+kind: reference
+sidebar: "language:lambdacalc"
+prev: /languages/lambdacalc/
+languages: [lambdacalc]
+tags:
+  - testing
+---
+
+# lc-test
+
+To run tests for Lambda Calculus programs, Codewars uses a custom compiling and testing framework, published and available in [this GitHub repository][test-framework-repo].
+
+The framework is hosted in Javascript, and as such, both the sample and submission tests of Kata must be written in Javascript, while the preloaded section, example solution and user solution are Lambda Calculus.
+
+Please refer to specific documentation for further information on the modules which `lc-test` utilizes.
+- [Chai](chai-docs) for assertions
+- [Lambda Calculus](lc-docs) for compilation
+
+## Basic Usage
+
+### Example tests
+```javascript
+import { assert, config, LC, getSolution } from "./lc-test.js";
+
+LC.configure({purity: "Let", numEncoding: "Church"});
+const { multiply } = LC.compile(getSolution());
+
+describe("Fixed tests", () => {
+  it("can multiply church numbers", () => {
+    assert.numEql(multiply(5)(0), 0);
+    assert.numEql(multiply(1)(8), 8);
+    assert.numEql(multiply(7)(9), 63);
+  });
+});
+```
+
+### Components
+
+`lc-test` provides the following utilities:
+* `chai.assert` as `assert`
+* `chai.config` as `config`
+* `LC = { compile, configure, fromInt, toInt }`
+* `getSolution`, `getPreloaded` and `getSolutionWithPreloaded`
+
+#### `assert`
+
+`assert` is the standard `chai.assert`, extended with the following the following additional assertion:
+
+**`assert.numEql(actual, expected, msg)`**
+
+`actual` and `expected` are automatically decoded according to the current `numEncoding`, if necessary, before making an equality assertion between them. A custom failure message can be set with `msg`.
```

Comment:
.. can _optionally_ be set ..

( Why not just write out `message` ? Doesn't cost anything. )

Reply:
Actually it would probably just throw an error about an invalid number, which would be caught by the `it` block. So i think it is fine.